### ISSUE TITLE
Metadata and conditional compilation breaking auto-complete

### ISSUE THREAD

**Issue by JonnyCodes** [HaxeFoundation/intellij-haxe#121]
Certain combinations of metadata and conditional compilation flags breaks auto-complete.

*\* BROKEN **
Metadata with conditional compilation flag after
   @:final#if flash@:keep #end
or
   @:final
   #if nape_swc@:keep #end

2 condition compilation flags on different lines
   #if flash dosomthing #end
   #if mobile dosomethingelse #end

*\* WORKING **
Metadata with condition compilation flag before
   #if nape_swc@:keep #end
   @:final

2 conditional compilation flags on the same line
   #if flash dosomthing #end #if mobile dosomethingelse #end

TIR: STB-7736

**Comment by as3boyan** [comment]
Maybe we can provide completion from compiler when code can't be parsed.

**Comment by EBatTiVo** [comment]
We need to provide completion from the compiler /always/ (issue #9).  Why would we do so only when there is a parsing error?

At any rate, we have plenty of errors with #if/#elsif/#end all over the place.  We can't just allow them between every keyword in the BNF, either.  So, we need a better solution: I'm thinking either just bite the bullet and get #9 and #7 done, or implement the ability to process and skip them in a pre-parsing phase (though I'm not sure how to do that yet).  And we still don't have a good strategy for dealing with errors that occur with one code path (selected via PP symbol) and not another, let alone refactoring, etc..

**Comment by EricBishton** [comment]
Tested on 0.11.2.  The following all parses correctly:
```haxe
package ;
class TestIssue121 {
    public function new() {
    }

    @:final #if flash@:keep #end public function keepme() {}  // broken

    @:final
    #if nape_swc@:keep #end         // broken
    public function keepme2() {}

    public static function doSomething() {
        #if flash dosomthing #end               // broken
        #if mobile dosomethingelse #end         // broken

        #if !flash keepme(); #end

        #if flash dosomthing #end #if mobile dosomethingelse #end  // working
    }

    #if nape_swc @:keep #end        // working
    @:final public function keepme3() {}

}
```
Likely fixed in the 0.11.0 cycle.  Closing.
